File: TP_lab7/TP_lab7_1/TP_lab7_1/ViewController.m
//
//  ViewController.m
//  TP_lab7_1
//
//  Created by Admin on 20.04.17.
//  Copyright © 2017 Daniil Boyko. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()

@property (weak, nonatomic) IBOutlet UIImageView *canvas;
@property CGPoint lastPoint;
@property (weak, nonatomic) IBOutlet UITextField *fieldRed;
@property (weak, nonatomic) IBOutlet UITextField *fieldGreen;
@property (weak, nonatomic) IBOutlet UITextField *fieldBlue;
@property (weak, nonatomic) IBOutlet UITextField *fieldWidth;
@property (weak, nonatomic) IBOutlet UITextField *fileField;

@property float Red,Green,Blue;

@end

@implementation ViewController


- (IBAction)setSettings:(id)sender {
    
}

- (void)viewDidLoad {
    [super viewDidLoad];
    

    // Do any additional setup after loading the view, typically from a nib.
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (IBAction)savePicture:(id)sender {
    UIImageWriteToSavedPhotosAlbum( _canvas.image,  nil, nil, nil);
}

- (void)touchesBegan:(NSSet *)touches withEvent:(UIEvent *)event
{
    UITouch *touch = [touches anyObject];
    
    [self setLastPoint:[touch locationInView:self.canvas]];
}

- (void)touchesMoved:(NSSet *)touches withEvent:(UIEvent *)event
{
    UITouch *touch = [touches anyObject];
    
    CGPoint currentPoint = [touch locationInView:self.canvas];
    
    UIGraphicsBeginImageContext(self.canvas.frame.size);
    
    CGRect drawRect = CGRectMake(0.0f, 0.0f, self.canvas.frame.size.width,
                                 
                                 self.canvas.frame.size.height);
    
    [[[self canvas] image] drawInRect:drawRect];
    
    CGContextSetLineCap(UIGraphicsGetCurrentContext(), kCGLineCapRound);
    
    CGContextSetLineWidth(UIGraphicsGetCurrentContext(), [[self.fieldWidth text] floatValue]);
    
    CGContextSetRGBStrokeColor(UIGraphicsGetCurrentContext(), [[self.fieldRed text] floatValue] /256, [[self.fieldGreen text] floatValue]/256, [[self.fieldBlue text] floatValue]/256, 1.0f);
    
    CGContextBeginPath(UIGraphicsGetCurrentContext());
    
    CGContextMoveToPoint(UIGraphicsGetCurrentContext(), _lastPoint.x,
                         
                         _lastPoint.y);
    
    CGContextAddLineToPoint(UIGraphicsGetCurrentContext(), currentPoint.x,
                            
                            currentPoint.y);
    
    CGContextStrokePath(UIGraphicsGetCurrentContext());
    
    [[self canvas] setImage:UIGraphicsGetImageFromCurrentImageContext()];
    
    UIGraphicsEndImageContext();
    
    _lastPoint = currentPoint;
}

@end

File: TP_lab7/TP_lab7_4/TP_lab7_4/ViewController.swift
//
//  ViewController.swift
//  TP_lab7_4
//
//  Created by Admin on 28.04.17.
//  Copyright © 2017 Daniil Boyko. All rights reserved.
//

import UIKit

class ViewController: UIViewController {

    @IBOutlet weak var gestureIndicator: UILabel!
    override func viewDidLoad() {
        super.viewDidLoad()
        gestureIndicator.isUserInteractionEnabled = true
        gestureIndicator.textAlignment = NSTextAlignment.center
        gestureIndicator.numberOfLines = 2;
        gestureIndicator.text = "Используйтежесты в этой области"
        gestureIndicator.backgroundColor = UIColor.yellow
        // Do any additional setup after loading the view, typically from a nib.
    }

    @IBAction func tap(_ sender: Any)
    {
        gestureIndicator.text = "Жест: касание\n Цвет фона:зеленый"
        gestureIndicator.backgroundColor = UIColor.green
    }
    
    @IBAction func pinch(_ sender: Any) {
        gestureIndicator.text = "Жест: масштабирование\n Цвет фона:красный"
        gestureIndicator.backgroundColor = UIColor.red
    }
    
    @IBAction func rotation(_ sender: Any) {
        gestureIndicator.text = "Жест: вращение\n Цвет фона:синий"
        gestureIndicator.backgroundColor = UIColor.blue

    }
    
    @IBAction func swipe(_ sender: Any) {
        gestureIndicator.text = "Жест: смахивание\n Цвет фона: серый"
        gestureIndicator.backgroundColor = UIColor.lightGray

    }
    
    @IBAction func longPress(_ sender: Any) {
        gestureIndicator.text = "Жест: долгое нажатие\n Цвет фона:оранжевый"
        gestureIndicator.backgroundColor = UIColor.orange

    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }

}

